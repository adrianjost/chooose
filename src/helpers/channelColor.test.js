import {
	getCenterColor,
	getChannelsForHueAndBrightness,
	getColorForHueAndBrightness,
	getColorForChannels,
	getHueAndBrightnessForChannels,
} from "./channelColor";

const testColorA = "#fd9";
const testColorB = "#9df";

describe("channelColor.js", () => {
	describe("getCenterColor", () => {
		it("#fd9 #9df", () => {
			expect(getCenterColor("#fd9", "#9df")).toMatchSnapshot();
		});
		it("#ffdd99 #99ddff", () => {
			expect(getCenterColor("#ffdd99", "#99ddff")).toMatchSnapshot();
		});
		it("#f00 #00f", () => {
			expect(getCenterColor("#f00", "#00f")).toMatchSnapshot();
		});
	});

	describe("getChannelsForHueAndBrightness", () => {
		it("min Hue max Brightness", () => {
			expect(
				getChannelsForHueAndBrightness({ hue: 0, brightness: 1 })
			).toMatchSnapshot();
		});
		it("mid Hue max Brightness", () => {
			expect(
				getChannelsForHueAndBrightness({ hue: 0.5, brightness: 1 })
			).toMatchSnapshot();
		});
		it("max Hue max Brightness", () => {
			expect(
				getChannelsForHueAndBrightness({ hue: 1, brightness: 1 })
			).toMatchSnapshot();
		});

		it("min Hue mid Brightness", () => {
			expect(
				getChannelsForHueAndBrightness({ hue: 0, brightness: 0.5 })
			).toMatchSnapshot();
		});
		it("mid Hue mid Brightness", () => {
			expect(
				getChannelsForHueAndBrightness({ hue: 0.5, brightness: 0.5 })
			).toMatchSnapshot();
		});
		it("max Hue mid Brightness", () => {
			expect(
				getChannelsForHueAndBrightness({ hue: 1, brightness: 0.5 })
			).toMatchSnapshot();
		});

		it("min Hue min Brightness", () => {
			expect(
				getChannelsForHueAndBrightness({ hue: 0, brightness: 0 })
			).toMatchSnapshot();
		});
		it("mid Hue min Brightness", () => {
			expect(
				getChannelsForHueAndBrightness({ hue: 0.5, brightness: 0 })
			).toMatchSnapshot();
		});
		it("max Hue min Brightness", () => {
			expect(
				getChannelsForHueAndBrightness({ hue: 1, brightness: 0 })
			).toMatchSnapshot();
		});
	});

	describe("getColorForHueAndBrightness", () => {
		it("min Hue max Brightness", () => {
			expect(
				getColorForHueAndBrightness({
					hue: 0,
					brightness: 1,
					colorA: testColorA,
					colorB: testColorB,
				})
			).toMatchSnapshot();
		});
		it("mid Hue max Brightness", () => {
			expect(
				getColorForHueAndBrightness({
					hue: 0.5,
					brightness: 1,
					colorA: testColorA,
					colorB: testColorB,
				})
			).toMatchSnapshot();
		});
		it("max Hue max Brightness", () => {
			expect(
				getColorForHueAndBrightness({
					hue: 1,
					brightness: 1,
					colorA: testColorA,
					colorB: testColorB,
				})
			).toMatchSnapshot();
		});

		it("min Hue mid Brightness", () => {
			expect(
				getColorForHueAndBrightness({
					hue: 0,
					brightness: 0.5,
					colorA: testColorA,
					colorB: testColorB,
				})
			).toMatchSnapshot();
		});
		it("mid Hue mid Brightness", () => {
			expect(
				getColorForHueAndBrightness({
					hue: 0.5,
					brightness: 0.5,
					colorA: testColorA,
					colorB: testColorB,
				})
			).toMatchSnapshot();
		});
		it("max Hue mid Brightness", () => {
			expect(
				getColorForHueAndBrightness({
					hue: 1,
					brightness: 0.5,
					colorA: testColorA,
					colorB: testColorB,
				})
			).toMatchSnapshot();
		});

		it("min Hue min Brightness", () => {
			expect(
				getColorForHueAndBrightness({
					hue: 0,
					brightness: 0,
					colorA: testColorA,
					colorB: testColorB,
				})
			).toMatchSnapshot();
		});
		it("mid Hue min Brightness", () => {
			expect(
				getColorForHueAndBrightness({
					hue: 0.5,
					brightness: 0,
					colorA: testColorA,
					colorB: testColorB,
				})
			).toMatchSnapshot();
		});
		it("max Hue min Brightness", () => {
			expect(
				getColorForHueAndBrightness({
					hue: 1,
					brightness: 0,
					colorA: testColorA,
					colorB: testColorB,
				})
			).toMatchSnapshot();
		});
	});

	describe("getColorForChannels", () => {
		it("[1, 1] with default colors", () => {
			expect(getColorForChannels([1, 0])).toMatchSnapshot();
		});

		it("[1, 0]", () => {
			expect(
				getColorForChannels([1, 0], testColorA, testColorB)
			).toMatchSnapshot();
		});
		it("[1, 1]", () => {
			expect(
				getColorForChannels([1, 1], testColorA, testColorB)
			).toMatchSnapshot();
		});
		it("[0, 1]", () => {
			expect(
				getColorForChannels([0, 1], testColorA, testColorB)
			).toMatchSnapshot();
		});
		it("[0.5, 0.25]", () => {
			expect(
				getColorForChannels([0.5, 0.25], testColorA, testColorB)
			).toMatchSnapshot();
		});
		it("[0.5, 0.5]", () => {
			expect(
				getColorForChannels([0.5, 0.5], testColorA, testColorB)
			).toMatchSnapshot();
		});
		it("[0.25, 0.5]", () => {
			expect(
				getColorForChannels([0.25, 0.5], testColorA, testColorB)
			).toMatchSnapshot();
		});
		it("[0, 0]", () => {
			expect(
				getColorForChannels([0, 0], testColorA, testColorB)
			).toMatchSnapshot();
		});
	});

	describe("getHueAndBrightnessForChannels", () => {
		it("[1, 0]", () => {
			expect(getHueAndBrightnessForChannels([1, 0])).toMatchSnapshot();
		});
		it("[1, 1]", () => {
			expect(getHueAndBrightnessForChannels([1, 1])).toMatchSnapshot();
		});
		it("[0, 1]", () => {
			expect(getHueAndBrightnessForChannels([0, 1])).toMatchSnapshot();
		});
		it("[0.5, 0.25]", () => {
			expect(getHueAndBrightnessForChannels([0.5, 0.25])).toMatchSnapshot();
		});
		it("[0.5, 0.5]", () => {
			expect(getHueAndBrightnessForChannels([0.5, 0.5])).toMatchSnapshot();
		});
		it("[0.25, 0.5]", () => {
			expect(getHueAndBrightnessForChannels([0.25, 0.5])).toMatchSnapshot();
		});
		it("[0.25, 0]", () => {
			expect(getHueAndBrightnessForChannels([0.25, 0])).toMatchSnapshot();
		});
		it("[0, 0]", () => {
			expect(getHueAndBrightnessForChannels([0, 0])).toMatchSnapshot();
		});
		it("[0, 0.25]", () => {
			expect(getHueAndBrightnessForChannels([0, 0.25])).toMatchSnapshot();
		});
	});
});
