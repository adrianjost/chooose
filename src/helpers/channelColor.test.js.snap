// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`channelColor.js getCenterColor #f00 #00f 1`] = `"#ff00ff"`;

exports[`channelColor.js getCenterColor #fd9 #9df 1`] = `"#ffffff"`;

exports[`channelColor.js getCenterColor #ffdd99 #99ddff 1`] = `"#ffffff"`;

exports[`channelColor.js getChannelsForHueAndBrightness max Hue max Brightness 1`] = `
Array [
  0,
  1,
]
`;

exports[`channelColor.js getChannelsForHueAndBrightness max Hue mid Brightness 1`] = `
Array [
  0,
  0.5,
]
`;

exports[`channelColor.js getChannelsForHueAndBrightness max Hue min Brightness 1`] = `
Array [
  0,
  0,
]
`;

exports[`channelColor.js getChannelsForHueAndBrightness mid Hue max Brightness 1`] = `
Array [
  1,
  1,
]
`;

exports[`channelColor.js getChannelsForHueAndBrightness mid Hue mid Brightness 1`] = `
Array [
  0.5,
  0.5,
]
`;

exports[`channelColor.js getChannelsForHueAndBrightness mid Hue min Brightness 1`] = `
Array [
  0,
  0,
]
`;

exports[`channelColor.js getChannelsForHueAndBrightness min Hue max Brightness 1`] = `
Array [
  1,
  0,
]
`;

exports[`channelColor.js getChannelsForHueAndBrightness min Hue mid Brightness 1`] = `
Array [
  0.5,
  0,
]
`;

exports[`channelColor.js getChannelsForHueAndBrightness min Hue min Brightness 1`] = `
Array [
  0,
  0,
]
`;

exports[`channelColor.js getColorForChannels [0, 0] 1`] = `"#000000"`;

exports[`channelColor.js getColorForChannels [0, 1] 1`] = `"#99ddff"`;

exports[`channelColor.js getColorForChannels [0.5, 0.5] 1`] = `"#808080"`;

exports[`channelColor.js getColorForChannels [0.5, 0.25] 1`] = `"#807766"`;

exports[`channelColor.js getColorForChannels [0.25, 0.5] 1`] = `"#667780"`;

exports[`channelColor.js getColorForChannels [1, 0] 1`] = `"#ffdd99"`;

exports[`channelColor.js getColorForChannels [1, 1] 1`] = `"#ffffff"`;

exports[`channelColor.js getColorForChannels [1, 1] with default colors 1`] = `"#ffdd99"`;

exports[`channelColor.js getColorForHueAndBrightness max Hue max Brightness 1`] = `"#99ddff"`;

exports[`channelColor.js getColorForHueAndBrightness max Hue mid Brightness 1`] = `"#4d6f80"`;

exports[`channelColor.js getColorForHueAndBrightness max Hue min Brightness 1`] = `"#000000"`;

exports[`channelColor.js getColorForHueAndBrightness mid Hue max Brightness 1`] = `"#ffffff"`;

exports[`channelColor.js getColorForHueAndBrightness mid Hue mid Brightness 1`] = `"#808080"`;

exports[`channelColor.js getColorForHueAndBrightness mid Hue min Brightness 1`] = `"#000000"`;

exports[`channelColor.js getColorForHueAndBrightness min Hue max Brightness 1`] = `"#ffdd99"`;

exports[`channelColor.js getColorForHueAndBrightness min Hue mid Brightness 1`] = `"#806f4d"`;

exports[`channelColor.js getColorForHueAndBrightness min Hue min Brightness 1`] = `"#000000"`;

exports[`channelColor.js getHueAndBrightnessForChannels [0, 0.25] 1`] = `
Object {
  "brightness": 0.25,
  "hue": 1,
}
`;

exports[`channelColor.js getHueAndBrightnessForChannels [0, 0] 1`] = `
Object {
  "brightness": 0,
  "hue": 0,
}
`;

exports[`channelColor.js getHueAndBrightnessForChannels [0, 1] 1`] = `
Object {
  "brightness": 1,
  "hue": 1,
}
`;

exports[`channelColor.js getHueAndBrightnessForChannels [0.5, 0.5] 1`] = `
Object {
  "brightness": 0.5,
  "hue": 0.5,
}
`;

exports[`channelColor.js getHueAndBrightnessForChannels [0.5, 0.25] 1`] = `
Object {
  "brightness": 0.5,
  "hue": 0.25,
}
`;

exports[`channelColor.js getHueAndBrightnessForChannels [0.25, 0.5] 1`] = `
Object {
  "brightness": 0.5,
  "hue": 0.75,
}
`;

exports[`channelColor.js getHueAndBrightnessForChannels [0.25, 0] 1`] = `
Object {
  "brightness": 0.25,
  "hue": 0,
}
`;

exports[`channelColor.js getHueAndBrightnessForChannels [1, 0] 1`] = `
Object {
  "brightness": 1,
  "hue": 0,
}
`;

exports[`channelColor.js getHueAndBrightnessForChannels [1, 1] 1`] = `
Object {
  "brightness": 1,
  "hue": 0.5,
}
`;
